ruby/ql/test/query-tests/variables/UninitializedLocal/UninitializedLocal.rb

```diff
@@ -0,0 +1,37 @@
+def m
+    puts "m"
+end
+
+def foo
+    m # calls m above
+    if false
+        m = 0
+        m # reads local variable m
+    else
+    end
+    m  #$ Alert
+    m2 # undefined local variable or method 'm2' for main (NameError)
+end
+
+def test_guards
+    if (a = 3 && a)  #$ SPURIOUS: Alert
+        a
+    end
+    if false
+        b = 0
+    end
+    b.nil?
+    b || 0  # OK
+    b&.m  # OK - safe navigation
+end
+
+def test_loop
+    begin
+        if false
+            a = 0
+        else
+            set_a
+        end   
+    end until a  # OK
```

Comment:
So the reason that this is OK is because `nil` is treated as `false` in a Boolean context?

Reply:
Yes, this loop will break once `a` is not `nil`.